lib/spack/spack/build_systems/cmake.py

```diff
@@ -246,3 +256,79 @@ def check(self):
 
     # Check that self.prefix is there after installation
     run_after('install')(PackageBase.sanity_check_prefix)
+
+    def write_spconfig(self, spack_env, spconfig_fname, dirty):
+        fname = os.path.join(spack_env.path, spconfig_fname) \
+            if spack_env is not None else spconfig_fname
+
+        with open(fname, 'w') as fout:
+            self._write_spconfig(fout, dirty)
+            fout.write('\nproc = subprocess.Popen(cmd, env=env)\n'
+                       'proc.wait()\n')
+
+        set_executable(fname)
+        return fname
+
+    def _write_spconfig(self, fout, dirty):
+        """Writes the spconfig.py file to a stream."""
+
+        # Set-up the environment
+        _cmd = [str(which('cmake'))] + \
+            self.std_cmake_args + self.cmake_args()
+
+        # No verbose makefile for interactive builds
+        cmd = [x for x in _cmd if not x.startswith('-DCMAKE_VERBOSE_MAKEFILE')]
+
+        env = dict()
+
+        paths = os.environ['PATH'].split(':')
+        paths = [item for item in paths if 'spack/env' not in item]
+        env['PATH'] = ':'.join(paths)
+        env['SPACK_TRANSITIVE_INCLUDE_PATH'] = spack_transitive_include_path()
```

Comment:
> What is the purpose of this variable? [SPACK_TRANSITIVE_INCLUDE_PATH]

See: https://spack.readthedocs.io/en/latest/workflows.html

In more detail: the case in which A -> B -> C and A includes B.h which includes C.h is not so uncommon.  This works OK in the pre-Spack world where everything is installed in a single directory tree (or if someone uses Spack Views).  But in a "clean" world with each package in its own directory, packages that used to build pre-Spack can end up not working.

I suppose the "right" way is to require A to list C as a dependency.  And since that's done in the Spack package (not the upstream code), it IS something we can require if we like.

What is `SPACK_DEPENDENCIES`?  It occurs 14 times in the Spack code, but not in the docs.

Reply:
OK I convinced myself to get rid of `SPACK_TRANSITIVE_INCLUDE_PATH`.  It can always be added back later, of course.  (@scheibelp Do the individual commits on this PR get archived anywhere after it is merged?)